Duplicati/Library/Backend/GoogleServices/GoogleDrive.cs

```diff
@@ -253,10 +307,10 @@ public IEnumerable<IFileEntry> List()
             }
         }
 
-        public async Task PutAsync(string remotename, string filename, CancellationToken cancelToken)
+        public Task PutAsync(string remotename, string filename, CancellationToken cancelToken)
         {
             using (System.IO.FileStream fs = System.IO.File.OpenRead(filename))

```

Comment:
I don't think this works. The using statement will dispose the file stream when exiting the scope (returning the task). At this point the task may not have completed, and an error will occur when the code tries to process the disposed filestream. Using `await` inside the using will generate a state machine that will defer the dispose call until after the `PutAsync` operation has completed.